vault/cluster.go

```diff
@@ -23,21 +23,23 @@ type Cluster struct {
 }
 
 // Cluster fetches the details of either local or global cluster based on the
-// input. This method errors out when Vault is sealed.
+// input. This method errors out when Vault is sealed. This function never
+// returns a nil Cluster object.
 func (c *Core) Cluster() (*Cluster, error) {
+	var cluster Cluster
+
 	// Fetch the storage entry. This call fails when Vault is sealed.
 	entry, err := c.barrier.Get(coreLocalClusterInfoPath)
 	if err != nil {
-		return nil, err
+		return &cluster, err
```

Comment:
I actually think that this should return nil. On errors, we want the error to be handled; if the error isn't being handled, trying to operate on a nil pointer is a good way to ensure that people realize that they aren't handling the error (assuming it's caught in tests :-D )